### PR TITLE
Add a script to report locationText for the position of the system caret

### LINKED ISSUE #9147: Add alternative way to report name of the current Excel sheet which don't require moving review cursor into Excel window
### Is your feature request related to a problem? Please describe.
I am often working with Excel workbooks containing multiple sheets and at the same time reading some instructions or data in another program with review cursor to avoid moving back and forth between two windows. In this case there is no easy way for me to determine in which sheet I am, because NVDA+ numpad delete works only when review cursor is in the Excel window.
### Describe the solution you'd like
There should be a command (NVDA+Alt+s perhaps) to announce name of the current sheet. 
### Describe alternatives you've considered
NVDA+tab might eventually announce current sheet name or maybe NVDA+t.
### Additional context
For the sighted users the sheet name is visible regardless of how many windows they have open, in which window they mouse cursor is if the Excel window is visible. Furthermore we are interested in the sheet in which we have focus not in the sheet in which our review cursor is similar to NVDA+tab. I understand that in the default config they are in the same place, but if we aren't accounting for the users which are using it separately why even create options to make them separate.

### PR DESCRIPTION
**Pull request by lukaszgo1**
### Link to issue number:
Closes #9147
Related to #7314

### Summary of the issue:
Currently it is impossible to report locationText for the current caret position only for the position of the review cursor. This is problematic in several cases:
- As described in #9147 there are cases in which it is useful to query  the name of the current sheet in MS Excel
- In edit controls and / or web browsers sighted people can simply glance at the scroll bar to determine their position in the text

While for people working with review cursor following system caret current NVDA+Numpad delete is sufficient for these use cases when review cursor does not follow caret you need  to move it to focus every time when you need to check your current position which wastes time, and more importantly makes your review position lost.

### Description of how this pull request fixes the issue:
- NVDA+numpad delete reports location of the caret or focused  object if there is no caret by default.
- Script for reporting review cursor position has been reassigned to NVDA+shift+Numpad delete - for most users there is no change as navigator follows focus and review follows caret by default.
- The modified scripts are splitted into separate ones for querying location of the text / object as discussed in #7314.
### Testing strategy:
Ensured that modified scripts works.
### Known issues with pull request:
None known
### Change log entries:
Changes
- NVDA+Numpad delete reports location of the caret / focused object by default - add shift to report location of the review cursor.

### Code Review Checklist:

- [X] Pull Request description:
  - description is up to date
  - change log entries
- [X] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [X] API is compatible with existing add-ons.
- [X] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [X] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

File: source/globalCommands.py [hunk 1]
```diff
@@ -1800,9 +1868,10 @@ def _reportFormattingHelper(self, info, browseable=False):
 			)
 
 	@staticmethod
-	def _getTIAtCaret(fallbackToPOSITION_FIRST=False):
+	def _getTIAtCaret(fallbackToPOSITION_FIRST=False, reportFailure=True):
```
Comment (seanbudd) [inline source/globalCommands.py:1871]:
```suggestion
	def _getTIAtCaret(fallbackToPOSITION_FIRST: bool = False, reportFailure: bool = True) -> None:
```

File: source/globalCommands.py [hunk 2]
```diff
@@ -1066,25 +1066,93 @@ def script_navigatorObject_current(self,gesture):
...
+			self.script_reportReviewCursorLocation(gesture)
+		else:
+			self.script_reportCurrentNavigatorObjectLocation(gesture)
+
+	@script(
+		description=_(
+			# Translators: Description for a keyboard command
+			# which reports location of the text at the caret position
+			# or object with focus if there is no caret.
+			"Reports information about the location of the text or object at the positionn of system caret "
```
Comment (seanbudd) [inline source/globalCommands.py:1145]:
```suggestion
			"Reports information about the location of the text or object at the position of system caret. "
```

File: source/globalCommands.py [hunk 3]
```diff
@@ -1066,25 +1066,93 @@ def script_navigatorObject_current(self,gesture):
...
+	)
+	def script_reportCurrentNavigatorObjectLocation(self, gesture):
+		self._reportLocationText((api.getNavigatorObject(),))
+
+	@script(
+		description=_(
+			# Translators: Description for a keyboard command which reports location of the
+			# current caret position falling back to the location of focused object if needed.
+			"Reports information about the location of the text at the caret, "
+			"or  location of the  currently focused object if there is no caret."
```
Comment (seanbudd) [inline source/globalCommands.py:1105]:
```suggestion
			"or location of the currently focused object if there is no caret."
```

File: source/globalCommands.py [hunk 4]
```diff
@@ -1066,25 +1066,93 @@ def script_navigatorObject_current(self,gesture):
 		else:
 			speech.speakObject(curObject, reason=controlTypes.OutputReason.QUERY)
 
+	@staticmethod
+	def _reportLocationText(objs) -> None:
```
Comment (seanbudd) [inline source/globalCommands.py:1070]: Could `objs` get a type hint?

File: source/globalCommands.py [hunk 5]
```diff
@@ -1066,25 +1066,93 @@ def script_navigatorObject_current(self,gesture):
...
+			if obj is not None and obj.locationText:
+				ui.message(obj.locationText)
+				return
+		# Translators: message when there is no location information
+		ui.message(_("No location information"))
+
+	@script(
+		description=_(
+			# Translators: Description for a keyboard command which reports location of the
+			# review cursor falling back to the location of navigator object if needed.
```
Comment (seanbudd) [inline source/globalCommands.py:1081]:
```suggestion
			# review cursor, falling back to the location of navigator object if needed.
```

File: source/globalCommands.py [hunk 6]
```diff
@@ -1066,25 +1066,93 @@ def script_navigatorObject_current(self,gesture):
...
+			"Reports information about the location of the text at the review cursor "
+			"or  location of the  navigator object if there is no text under review cursor."
```
Comment (seanbudd) [inline source/globalCommands.py:1083]:
```suggestion
			"or location of the navigator object if there is no text under review cursor."
```

**Review by seanbudd — CHANGES_REQUESTED** [review]
Thanks @lukaszgo1, generally looks good to me. I've added a few whitespace and other minor fixes.

**Comment by lukaszgo1** [comment]
@seanbudd  I believe all changes you've requested are now addressed.

**Review by seanbudd — APPROVED** [review]
Thanks @lukaszgo1, looks good to me.
Going to wait for @Qchristensen to check the user guide changes before merging.

**Review by Qchristensen — APPROVED** [review]
Looks good, great work!

**Pull request merged**